src/main/java/perkmanager/PerksController.java

```diff
@@ -95,4 +95,16 @@ public String viewMembershipPerks(@PathVariable("id") Long membershipId, Model m
         // return view name for Thymeleaf fragment
         return "allPerks :: content";
     }
+
+    @GetMapping("/searchPerk/name/{perkName}")
+    public String viewMembershipPerks(@PathVariable("perkName") String perkName, Model model) {
+        List<Perk> perks = new ArrayList<>();
+        Perk perk = membershipService.findByPerkName(perkName);
```

Comment:
Can you make it so it does not have to be the exact name? You can iterate over all the perks and check if what the user searched for is a substring of the perk name, then add it to the list of perks.

Reply:
I can definitely do that! I will make that change and commit again.